### ISSUE TITLE
[Feature request] - update python packaging

### ISSUE THREAD

**Issue by shimwell** [JADE-V-V/JADE#294]
I've been looking through a few F4E repos recently and I see most have setup.py or setup.cfg files. I noticed this one also has a partial pyproject.toml file. We could move these setup files entirely to pyproject.toml files to keep up to date with the latest python packaging recommendations
- [https://packaging.python.org/en/latest/guides/modernize-setup-py-project/#modernize-setup-py-project](https://packaging.python.org/en/latest/guides/modernize-setup-py-project/#modernize-setup-py-project)
- [https://packaging.python.org/en/latest/tutorials/packaging-projects/](https://packaging.python.org/en/latest/tutorials/packaging-projects/)

While updating we could also move to a ```src/jade``` layout instead of the ```jade``` folder structure. The version could be moved to the pyprojects and we could make use of ```import importlib.metadata``` to make the ```jade.__version__``` attribute or if we are really keen then we could make use of the setuptools_scm to track the version for us.

**Describe the solution you'd like**
- pyproject.toml instead of setup.py + setup.cfg
- src/jade instead of jade folder structure
- importlib.metadata or setuptools_scm to get version number

**Describe alternatives you've considered**
could also leave things as they are

**Additional context**
while setup.py is not depreciated the command commonly associated with a setup.py file ```python setup.py install``` is now depreciated and pyproject.toml is "strongly recommened" for python packaging

**Comment by dodu94** [comment]
Due to the number of times we are forgetting to update the version number before release I'm starting to think we should find the time to do this with a quick "git mv" and then implement dynamic versioning

**Comment by Radiation-Transport** [comment]
I agree on dynamic versioning for python. May we use something similar for the versioning of JADE (for the promp header for instance)?

**Comment by dodu94** [comment]
I was referring to dynamic versioning for JADE. The prompt header (and other strings) are already all connected to a single variable in the __version__.py file. The problem is that at the moment we are updating this constant by hand.
